x-pack/legacy/plugins/ml/public/application/jobs/new_job/pages/components/pick_fields_step/components/bucket_span_estimator/bucket_span_estimator.tsx

```diff
@@ -19,18 +21,37 @@ export const BucketSpanEstimator: FC<Props> = ({ setEstimating }) => {
   const { jobCreator, jobCreatorUpdate } = useContext(JobCreatorContext);
   const { status, estimateBucketSpan } = useEstimateBucketSpan();
   const [noDetectors, setNoDetectors] = useState(jobCreator.detectors.length === 0);
+  const [isUsingMlCategory, setIsUsingMlCategory] = useState(checkIsUsingMlCategory());
 
   useEffect(() => {
     setEstimating(status === ESTIMATE_STATUS.RUNNING);
   }, [status]);
 
   useEffect(() => {
     setNoDetectors(jobCreator.detectors.length === 0);
+    setIsUsingMlCategory(checkIsUsingMlCategory());
   }, [jobCreatorUpdate]);
 
+  function checkIsUsingMlCategory() {
+    return (
+      isAdvancedJobCreator(jobCreator) &&
+      jobCreator.detectors.some(d => {
+        if (
+          d.partition_field_name === MLCATEGORY ||
+          d.over_field_name === MLCATEGORY ||
+          d.by_field_name === MLCATEGORY
+        ) {
+          return true;
+        }
+      })
+    );
+  }
+
   return (
     <EuiButton
-      disabled={status === ESTIMATE_STATUS.RUNNING || noDetectors === true}
+      disabled={
```

Comment:
Ideally it would be nice to have a tooltip saying why the estimate bucket span is disabled, as otherwise it is not obvious why the button switches to being disabled.

Reply:
yeah, i agree. Could this be done in a follow up PR?